packages/snap/src/handlers/set-network.tsx

```diff
@@ -44,8 +53,25 @@ export const setNetwork: Handler<SetNetworkParams, SetNetworkResponse> = async (
   const publicApi = new PublicApiClient(config);
   const node_status = await publicApi.getNodeStatus();
 
+  const networkName = CHAIN_ID_TO_NETWORK_NAME[node_status.chain_id.toString()];
+
+  const confirm = await snap.request({
+    method: 'snap_dialog',
+    params: {
+      type: 'confirmation',
+      content: (
+        <SwitchNetwork networkName={networkName || ''} rpcUrl={network} />
+      ),
+    },
+  });
+
+  if (!confirm) {
+    throw new Error('User denied calling smart contract');
+  }
+
   await setActiveRPC(network);
   await setActiveChainId(node_status.chain_id);
+  await setActiveMinimalFees(node_status.minimal_fees || DEFAULT_MINIMAL_FEES);
```

Comment:
prefere the use of ?? instead ||
here if `node_status.minimal_fees` is 0 it will take `DEFAULT_MINIMAL_FEES`and its not what we want